test/test_trainer.py

```diff
@@ -132,6 +132,24 @@ def test_classification_trainer(self):
             # Check that params were updated
             self.assertFalse(torch.equal(model[-1].weight.data, model_w))
 
+    def test_binary_classification_trainer(self):
+
+        num_it = 10
+        batch_size = 8
+        # Generate all dependencies
+        model = nn.Sequential(nn.Conv2d(3, 32, 3), nn.ReLU(inplace=True),
+                              GlobalAvgPool2d(flatten=True), nn.Linear(32, 1))
+        train_loader = torch.utils.data.DataLoader(MockClassificationDataset(num_it * batch_size),
+                                                   batch_size=batch_size)
+        optimizer = torch.optim.Adam(model.parameters())
+        criterion = torch.nn.CrossEntropyLoss()
+
+        learner = trainer.BinaryClassificationTrainer(model, train_loader, train_loader, criterion, optimizer)
```

Comment:
There is an issue here: if this is for binary classification, it needs a proper loss (BCE rather than Cross entropy), and a proper dataset (MockClassificationDataset is for multi-class classification).

Reply:
I don't get your point here, MockClassificationDataset returns an image and a target to 0. 0 can either be a class identifier or a probability.

Reply:
ok I get it know, is about the size of the output tensor :/